src/jit/optimizer.cpp

```diff
@@ -2834,365 +2829,373 @@ void Compiler::optUnrollLoops()
-                    /* Get the expression and stop if end reached */
-
-                    GenTreePtr expr = stmt->gtStmtExpr;
-                    if (expr == incr)
-                    {
-                        break;
-                    }
-
-                    /* Calculate gtCostSz */
-                    gtSetStmtInfo(stmt);
-
-                    /* Update loopCostSz */
-                    loopCostSz += stmt->gtCostSz;
-                }
-            } while (block != bottom);
+                break;
+            }
+        }
 
-            /* Compute the estimated increase in code size for the unrolled loop */
+        /* Compute the estimated increase in code size for the unrolled loop */
 
-            unsigned int fixedLoopCostSz;
-            fixedLoopCostSz = 8;
+        unsigned int fixedLoopCostSz;
+        fixedLoopCostSz = 8;
 
-            int unrollCostSz;
-            unrollCostSz = (loopCostSz * totalIter) - (loopCostSz + fixedLoopCostSz);
+        int unrollCostSz;
+        unrollCostSz = (loopCostSz * totalIter) - (loopCostSz + fixedLoopCostSz);
 
-            /* Don't unroll if too much code duplication would result. */
+        /* Don't unroll if too much code duplication would result. */
 
-            if (unrollCostSz > unrollLimitSz)
-            {
-                /* prevent this loop from being revisited */
-                optLoopTable[lnum].lpFlags |= LPFLG_DONT_UNROLL;
-                goto DONE_LOOP;
-            }
+        if (unrollCostSz > unrollLimitSz)
+        {
+            /* prevent this loop from being revisited */
+            optLoopTable[lnum].lpFlags |= LPFLG_DONT_UNROLL;
+            goto DONE_LOOP;
+        }
 
-            /* Looks like a good idea to unroll this loop, let's do it! */
-            CLANG_FORMAT_COMMENT_ANCHOR;
+        /* Looks like a good idea to unroll this loop, let's do it! */
+        CLANG_FORMAT_COMMENT_ANCHOR;
 
 #ifdef DEBUG
-            if (verbose)
+        if (verbose)
+        {
+            printf("\nUnrolling loop BB%02u", head->bbNext->bbNum);
+            if (head->bbNext->bbNum != bottom->bbNum)
             {
-                printf("\nUnrolling loop BB%02u", head->bbNext->bbNum);
-                if (head->bbNext->bbNum != bottom->bbNum)
-                {
-                    printf("..BB%02u", bottom->bbNum);
-                }
-                printf(" over V%02u from %u to %u", lvar, lbeg, llim);
-                printf(" unrollCostSz = %d\n", unrollCostSz);
-                printf("\n");
+                printf("..BB%02u", bottom->bbNum);
             }
+            printf(" over V%02u from %u to %u", lvar, lbeg, llim);
+            printf(" unrollCostSz = %d\n", unrollCostSz);
+            printf("\n");
+        }
 #endif
 
-            /* Create the unrolled loop statement list */
-
-            loopList = loopLast = nullptr;
+        /* Create the unrolled loop statement list */
+        {
+            BlockToBlockMap blockMap(getAllocator());
```

Comment:
It may be worth using a `SmallHashTable` here to avoid heap allocations for the map elements.